### PR TITLE
Remove reader code, support Python version 3.12, drop 3.8

### PR DESCRIPTION
**Pull request by mkundu1**
1. Move reader code to separate repo https://github.com/ansys/pyfluent-reader.
2. Support Python version 3.12, drop support for 3.8.

### CODE REVIEW

File: tests/test_rp_vars.py [hunk 1]
```diff
@@ -44,9 +43,10 @@ def test_get_all_rp_vars(new_solver_session_no_transcript) -> None:
     assert len(all_vars) == pytest.approx(9000, 20)
 
     # CaseFile comparison, note that the PyFluent work dir is not necessarily the same as the Fluent work dir
-    case = CaseReader(case_filepath=path(case_path))
-    case_vars = case.rp_vars()
-    assert len(case_vars) == pytest.approx(9000, 450)
+    # To be moved to reader repo
```
Comment (mkundu1) [inline tests/test_rp_vars.py:46]: https://github.com/ansys/pyfluent-reader/issues/9

**Review by hpohekar — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -365,7 +365,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.8
+          python-version: 3.9
```
Comment (RobPasMue) [inline .github/workflows/ci.yml:368]: @mkundu1 - have a look at other repos. We typically define a MAIN_PYTHON_VERSION. See https://github.com/ansys/pyprimemesh/blob/main/.github/workflows/ci_cd.yml#L19

That way it is easier to maintain and to see what is the main Python version you are doing operations against.

**Comment by seanpearsonuk** [comment]
@mkundu1 do we definitely want to break up the package at this point? I'm not against it per se, but if it's to support 3.12, should we wait, given that we have 6 mo soft deadline?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (prmukherj) [reply .github/workflows/ci.yml]: Done, Thank you.

**Comment by mkundu1** [comment]
> @mkundu1 do we definitely want to break up the package at this point? I'm not against it per se, but if it's to support 3.12, should we wait, given that we have 6 mo soft deadline?

Yeah, we shall wait at least till pyfluent-reader package is released.

**Comment by mkundu1** [comment]
closing this, will reactivate once we are ready to merge

**Pull request closed without merging**